code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -622,27 +699,19 @@
 	cyborg_base_icon = "miner"
 	moduleselect_icon = "miner"
 	hat_offset = 0
-	var/obj/item/t_scanner/adv_mining_scanner/cyborg/mining_scanner //built in memes.
-
-/obj/item/robot_module/miner/be_transformed_to(obj/item/robot_module/old_module)
-	var/mob/living/silicon/robot/cyborg = loc
-	var/list/miner_icons = list(
-		"Lavaland Miner" = image(icon = 'icons/mob/robots.dmi', icon_state = "miner"),
-		"Asteroid Miner" = image(icon = 'icons/mob/robots.dmi', icon_state = "minerOLD"),
-		"Spider Miner" = image(icon = 'icons/mob/robots.dmi', icon_state = "spidermin")
+	borg_skins = list(
+		"Lavaland Miner" = list(SKIN_ICON_STATE = "miner", SKIN_TRANSFORM = TRUE),
+		"Asteroid Miner" = list(SKIN_ICON_STATE = "minerOLD", SKIN_LIGHT_KEY = "miner", SKIN_TRANSFORM = TRUE),
+		"Spider Miner" = list(SKIN_ICON_STATE = "spidermin"),
+		"Basic" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "robot_miner", SKIN_LIGHT_KEY = "robot"),
+		"Drone" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "drone_miner"),
+		"Heavy" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "heavy_miner"),
+		"Noble" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "noble_miner", SKIN_LIGHT_KEY = "noble_standard"),
+		"Sleek" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "sleek_miner", SKIN_LIGHT_KEY = "sleek_standard"),
+		"Tall" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "tall2_miner"),
+		"Marina" = list(SKIN_ICON = 'icons/mob/robots_extra.dmi', SKIN_ICON_STATE = "marina_miner")
 	)
-	var/miner_robot_icon = show_radial_menu(cyborg, cyborg, miner_icons, custom_check = CALLBACK(src, .proc/check_menu, cyborg, old_module), radius = 42, require_near = TRUE)
-	switch(miner_robot_icon)
-		if("Lavaland Miner")
-			cyborg_base_icon = "miner"
-		if("Asteroid Miner")
-			cyborg_base_icon = "minerOLD"
-			special_light_key = "miner"
-		if("Spider Miner")
-			cyborg_base_icon = "spidermin"
-		else
-			return FALSE
-	return ..()
+	var/obj/item/t_scanner/adv_mining_scanner/cyborg/mining_scanner //built in memes.
```

Comment:
?

Reply:
If you look at the full code, scanner is already there, it just moved down because I added borg_skins variable above it for organization sake.